pyproject.toml

```diff
@@ -71,49 +71,161 @@ line-length = 79
 indent-width = 4
 
 target-version = "py38"
+exclude = [
+  "CSV",
+  "ref.*?py",
+]
 
 [tool.ruff.lint]
-fixable = ["ALL"]
-preview = true
+fixable = []
+# preview = true
+# logger-objects = []
+task-tags = ["TODO", "FIXME", "XXX"]
 select = [
+  "A", # flake8-builtins
+  "ARG", # flake8-unused-arguments
+  "COM", # flake8-commas
+  "D", # pydocstyle
   "E", # pycodestyle
+  "E701", # multiple-statements-on-one-line-colon
+  "E722", # bare-except
   "F", # Pyflakes
+  "F401", # unused-import
+  "G", # flake8-logging-format
+  "I", # isort
+  "N", # pep8-naming
+  "NPY", # numpy-specific
+  "PL", # Pylint
+  "Q", # flake8-quotes
+  "UP", # pyupgrade
```

Comment:
I love using UP, and will continue to locally, but maybe not needed for the standard rule set? I could go either way on that. What do you think?

Reply:
I suggest using it. The first time I used was a big surprise for myself. I've been using Python for so long that I didn't realize how I was getting outdated. That should help us to be aware of how the Python syntax is evolving. It will make a difference once we bump the minimum Python version.

That said, whatever you both like, works for me.

Reply:
Agreed. Lets keep it